students/MichaelM/lesson5/mailroom_rev_try_catch.py

```diff
@@ -0,0 +1,258 @@
+                  f"\n" \
+                  f"Kind regards,\n" \
+                  f"The Minimalist Society"
+    elif letter_type == "new" and spam_type == "ty":
+        results = f"[Subject]: Thank you for your generous donation\n" \
+                  f"\n" \
+                  f"[Body]: {donor},\n" \
+                  f"Thank you for your generous donation. We are always grateful to greet new members of " \
+                  f"our ever expanding family.\n" \
+                  f"Your current and future beneficent contributions will go towards " \
+                  f"administering our clients with our services.\n" \
+                  f"Your initial contribution of {donation} illustrates an exceedingly benevolent nature.\n" \
+                  f"\n" \
+                  f"Kind regards,\n" \
+                  f"The Minimalist Society\n" \
+                  f"[End]"
+    elif letter_type == "existing_donor" and spam_type == "ty":
+        results = f"[Subject]: Once again, thank you for your generous donation\n" \
+                  f"\n" \
+                  f"[Body]: {donor},\n" \
+                  f"Thank you for your generous donation. Your beneficent contribution of {donation} will go " \
+                  f"towards administering our clients with our services.\n" \
+                  f"Our records show to date your generosity " \
+                  f"of {total_donation} illustrates an exceedingly benevolent nature.\n" \
+                  f"\n" \
+                  f"Kind regards,\n" \
+                  f"The Minimalist Society\n" \
+                  f"[End]"
+    return results
+
+
+def create_ty_letter():
+    """
+        takes user input and incorporates this input into a list of donors and returns text to the screen
+
+        Extended description:
+        If a donor (user_response) exists the function will take the input, add their current donation and print
+        to screen an email template containing the new data. If the donor does not exists the donor will be added
+        to the list and the user will be prompted for a donation amount after which the new data will be incorporated
+        into email and printed to screen.
+
+        12/1/14/2018 function modified to add try/except blocks and list comprehensions
+
+
+        Parameters:
+        my_int (int): integer to be padded
+
+        Returns:
+        result (string): a padded number
+
+        """
+    found_donor_count = 0
+    response_donor_name = ""
+    response_donation_amt = 0.0
+    sum_amt = 0.0
+    tmp_donor_list = list(list_donors())
+    user_response = input(
+        "-type 'list' to view a list of donors\n-type 'new' to  "
+        "add a donor to the list\n- type in an existing donor to add a new donation >")
+    if user_response.lower() == "list":
+        print()
+        # converted to list comprehension
+        [print(''.join([str(elem) for elem in row])) for row in list_donors()]
```

Comment:
this is considered an anti-pattern (print() function + comprehension, print function returns `None`), use comprehensions when you are capturing the result